cairo1-run/src/main.rs

```diff
@@ -562,13 +615,35 @@ fn create_entry_code(
     func: &Function,
     initial_gas: usize,
     proof_mode: bool,
+    args: &Vec<FuncArg>,
 ) -> Result<(Vec<Instruction>, Vec<BuiltinName>), Error> {
     let mut ctx = casm! {};
     // The builtins in the formatting expected by the runner.
     let (builtins, builtin_offset) = get_function_builtins(func);
     // Load all vecs to memory.
+    // Load all array args content to memory.
+    let mut array_args_data = vec![];
     let mut ap_offset: i16 = 0;
-    let after_vecs_offset = ap_offset;
+    for arg in args {
+        let FuncArg::Array(values) = arg else { continue };
+        array_args_data.push(ap_offset);
+        casm_extend! {ctx,
+            %{ memory[ap + 0] = segments.add() %}
+            ap += 1;
+        }
+        for (i, v) in values.iter().enumerate() {
+            let arr_at = (i + 1) as i16;
+            casm_extend! {ctx,
+                [ap + 0] = (v.to_bigint());
```

Comment:
Shouldn't this also increment `ap` after writing `v`?

Reply:
This is copied from the original code [here](https://github.com/starkware-libs/cairo/blob/e0411d11efd530ba5b456a7c801416e55e680c4b/crates/cairo-lang-runner/src/lib.rs#L419)